Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
 #import <vector>
 #import <CoreGraphics/CGContext.h>
 #import <CoreGraphics/CGGeometry.h>
+#import <CoreGraphics/CGDataProvider.h>
 #import <Foundation/NSData.h>
-#import <UIKit/UIImage.h>
+#import <LoggingNative.h>
+#import <CFRuntime.h>
+#import <CFBridgeUtilities.h>
+#import "D2DUtility.h"
 #import "CGColorSpaceInternal.h"
 #import "CGImageInternal.h"
-#import "_CGLifetimeBridgingType.h"
 #import "CGSurfaceInfoInternal.h"
 
-extern "C" {
-#import <png.h>
-};
+using namespace Microsoft::WRL;
 
-// This is the format libpng expects.
-struct _RGBA_swizzle {
-    BYTE r, g, b, a;
-};
+static const wchar_t *TAG = L"CGImage";
 
-// This is what comes out of pixman.
-struct _BGRA_swizzle {
-    BYTE b, g, r, a;
-};
+#define RETURN_RESULT_IF_NULL(input, result)                                   \
+  if (!input) {                                                                \
+    return result;                                                             \
+  }
 
-#include "LoggingNative.h"
+#define RETURN_RESULT_IF_FAILED(input, result)                                 \
```

Comment:
this is not present in result.h.

i'll move it there so it can be used by others.

---
In reply to: [84145558](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84145558) [](ancestors = 84145558)